rpcserver.go

```diff
@@ -7358,7 +7373,26 @@ func (r *rpcServer) SendPayment(req *tchrpc.SendPaymentRequest,
 			peerPubKey = &parsedKey
 		}
 
-		specifier := asset.NewSpecifierFromId(assetID)
+		var rpcSpecifier rfqrpc.AssetSpecifier
+
+		switch {
+		case specifier.HasId():
+			assetID := specifier.UnwrapIdToPtr()
+			rpcSpecifier = rfqrpc.AssetSpecifier{
+				Id: &rfqrpc.AssetSpecifier_AssetId{
+					AssetId: assetID[:],
+				},
+			}
+
+		case specifier.HasGroupPubKey():
+			groupKey := specifier.UnwrapGroupKeyToPtr()
+			groupKeyBytes := groupKey.SerializeCompressed()
+			rpcSpecifier = rfqrpc.AssetSpecifier{
+				Id: &rfqrpc.AssetSpecifier_GroupKey{
+					GroupKey: groupKeyBytes,
+				},
+			}
+		}
```

Comment:
Probably makes sense to extract this into a `marshalAssetSpecifier` function, as this seems to appear twice in the code now.